contracts/src/l2/rewaave/token_bridge.cairo

```diff
@@ -236,24 +269,61 @@ func handle_deposit{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_chec
     from_address : felt,
     l1_sender : felt,
     l2_recipient : felt,
-    l2_token_address : felt,
+    l2_token : felt,
     amount_low : felt,
     amount_high : felt,
+    block_number_low : felt,
+    block_number_high : felt,
+    l1_rewards_index_low : felt,
+    l1_rewards_index_high : felt,
 ):
     alloc_locals
     only_l1_handler(from_address_=from_address)
 
-    let amount = Uint256(low=amount_low, high=amount_high)
+    let amount_ = Uint256(low=amount_low, high=amount_high)
+    local amount : Wad = Wad(amount_)
+
+    let l1_rewards_index_ = Uint256(low=l1_rewards_index_low, high=l1_rewards_index_high)
+    local l1_rewards_index : Ray = Ray(l1_rewards_index_)
+
+    let block_number = Uint256(low=block_number_low, high=block_number_high)
+
+    with_attr error_message("High or low overflows 128 bit bound {amount_}"):
+        uint256_check(amount_)
+    end
 
-    with_attr error_message("High or low overflows 128 bit bound {amount}"):
-        uint256_check(amount)
+    with_attr error_message("High or low overflows 128 bit bound {l1_rewards_index}_"):
+        uint256_check(l1_rewards_index_)
     end
-    assert_not_zero(l2_token_address)
 
-    # Call mint on l2_token contract.
+    with_attr error_message("High or low overflows 128 bit bound {block_number}"):
+        uint256_check(block_number)
+    end
+
+    assert_not_zero(l2_token)
+
+    let (reward_token) = rewAAVE.read()
 
-    IERC20.mint(contract_address=l2_token_address, recipient=l2_recipient, amount=amount)
-    deposit_handled.emit(l2_token_address, l1_sender, l2_recipient, amount)
+    # handle the difference of the index at send and recieve
+    let (current_index) = IETHstaticAToken.get_rewards_index(l2_token)
+    let (le) = ray_le(current_index, l1_rewards_index)
+    if le == 1:
+        IETHstaticAToken.push_rewards_index(
+            contract_address=l2_token, block_number=block_number, rewards_index=l1_rewards_index
+        )
+    else:
+        let (amount_ray) = wad_to_ray(amount)
+        let (reward_diff) = ray_sub(l1_rewards_index, current_index)
+        let (reward_outstanding_ray) = ray_mul_no_rounding(reward_diff, amount_ray)
+        let (reward_outstanding) = ray_to_wad_no_rounding(reward_outstanding_ray)
+        IERC20.mint(reward_token, l2_recipient, reward_outstanding.wad)
```

Comment:
if I've understood correctly, `reward_diff` is a negative number here (since current_index > l1_rewards_index here). don't we want to define it as `ray_sub(current_index, l1_rewards_index)` instead?

Reply:
Yes, thank you for spotting this